extensions/native/recursion/src/fri/two_adic_pcs.rs

```diff
@@ -111,15 +110,19 @@ pub fn verify_two_adic_pcs<C: Config>(
     iter_zip!(
         builder,
         proof.commit_phase_commits,
+        proof.commit_pow_witnesses,
         betas,
         betas_squared,
         i_plus_one_arr
     )
     .for_each(|ptr_vec, builder| {
-        let [comm_ptr, beta_ptr, beta_sq_ptr, i_plus_one_ptr] = ptr_vec.try_into().unwrap();
+        let [comm_ptr, commit_pow_ptr, beta_ptr, beta_sq_ptr, i_plus_one_ptr] =
+            ptr_vec.try_into().unwrap();
 
         let comm = builder.iter_ptr_get(&proof.commit_phase_commits, comm_ptr);
+        let commit_pow = builder.iter_ptr_get(&proof.commit_pow_witnesses, commit_pow_ptr);
```

Comment:
for safety, I think we should `builder.assert_usize_eq(proof.commit_pow_witnesses.len(), log_max_height);`

I could see some argument that it is not quite necessary due to the witness check, but it seems easy and safer to just add the length check?